### PR TITLE
Prototype Builds

### PR DESCRIPTION
**Pull request by mokagio**
Internal ref pdnsEh-1D4-p2

Builds on top of #1769 to finally distribute the Prototype Build.

## To test

Download and launch the Prototype Build generated from this branch.

<table>
<tr>
<td>

52D60EA9-D8EE-4357-9944-6305C873761A_1_102_o

</td>
<td>

DDC1DE4A-1053-4F1F-BF40-ACF5AFA7F228_1_102_o

</td>

## Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to `CHANGELOG.md` if necessary. – N.A.
- [x] I have considered adding unit tests for my changes. – N.A.
- [x] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics. – N.A.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 500 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/Automattic/pocket-casts-ios/milestone/52">7.66 ❄️</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by pocketcasts** [comment]
<details><summary>Pocket Casts Prototype Build📲 You can test the changes from this Pull Request in <b>Pocket Casts Prototype Build</b> by scanning the QR code below to install the corresponding build.</summary>
<table>
<tr>
  <td rowspan='6' width='260px'></td>
  <td><b>App Name</b></td><td>Pocket Casts Prototype Build Pocket Casts Prototype Build</td>
</tr>
<tr><td><b>Build Number</b></td><td>pr1771-524b10e</td></tr>
<tr><td><b>Version</b></td><td>7.65</td></tr>
<tr><td><b>Bundle ID</b></td><td>au.com.shiftyjelly.podcasts.prototype</td></tr>
<tr><td><b>Commit</b></td><td>524b10e6541ca9fb90c0bb4f7936f262113ed9cd</td></tr>
<tr><td><b>App Center Build</b></td><td><a href='https://install.appcenter.ms/orgs/automattic/apps/Pocket-Casts-Prototype-Builds/releases/10'>Pocket Casts - Prototype Builds #10</a></td></tr>
</table>
<em>Automatticians: You can use our internal self-serve MC tool to give yourself access to App Center if needed.</em>
</details>

**Comment by mokagio** [comment]
**Update:** The issue had to do with an out of date dependency. It is now fixed as you can see in the comment above.

---


🤔 not sure what's wrong with the QR code...

image

File: .buildkite/commands/build.sh [hunk 1]
```diff
@@ -1,9 +1,6 @@
 #!/bin/bash -u
 
 echo "--- Set up SPM"
-# We'd like to use this, but it doesn't work yet. See
-# https://github.com/Automattic/bash-cache-buildkite-plugin/issues/20 ...
-#
```
Comment (mokagio) [inline .buildkite/commands/build.sh:6]: This was an outdated leftover.

File: Podfile [hunk 2]
```diff
@@ -35,10 +35,10 @@ abstract_target 'CI' do
...
 end
 
-post_install do |pi|
-  pi.pods_project.targets.each do |t|
-    t.build_configurations.each do |config|
-      if t.name != 'Pocket Casts Watch App'
+post_install do |installer|
+  installer.pods_project.targets.each do |target|
+    target.build_configurations.each do |config|
+      if target.name != 'Pocket Casts Watch App'
```
Comment (mokagio) [inline Podfile:41]: The true reason for this refactor is to update the `Podfile.lock` so that CI will generate a new cached `Pods/` archive which will include the `xcconfig` for the new Prototype build configuration.

Otherwise, CI will download a `Pods/` folder version that does not include the files for that new build configuration and will fail to build with errors such as:

> error: Unable to load contents of file list: '/Target Support Files/Pods-podcasts/Pods-podcasts-frameworks-Prototype-output-files.xcfilelist' (in target 'podcasts' from project 'podcasts')

See
https://buildkite.com/automattic/pocket-casts-ios/builds/7090#018fe6b5-508b-443b-abb9-064a04ae0054

File: fastlane/Fastfile [hunk 3]
```diff
@@ -1128,9 +1151,9 @@ inal copy, and try again.")
   #
   def build_code_current
     # Read the current build code from the .xcconfig file and parse it into an AppVersion object
-    # The AppVersion is used because WCiOS uses the four part (1.2.3.4) build code format, so the version
+    # The AppVersion is used because Pocket Casts iOS uses the four part (1.2.3.4) build code format, so the version
```
Comment (mokagio) [inline fastlane/Fastfile:1175]: 🙈

File: podcasts.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved [hunk 4]
```diff
@@ -2,30 +2,264 @@
   "object": {
     "pins": [
       {
-        "package": "CryptoSwift",
-        "repositoryURL": "https://github.com/krzyzanowskim/CryptoSwift.git",
+        "package": "abseil",
+        "repositoryURL": "https://github.com/google/abseil-cpp-binary.git",
```
Comment (mokagio) [inline podcasts.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved:6]: I had to update the resolved file otherwise the build would fail.

Updated via

```
xcodebuild -resolvePackageDependencies -workspace podcasts.xcworkspace \
  -scheme pocketcasts -configuration Debug
```

Without this, the build will fail when fetching the packages with:

```
an out-of-date resolved file was detected at /opt/ci/builds/builder/automattic/pocket-casts-ios/podcasts.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved, which is not allowed when automatic dependency resolution is disabled; please make sure to update the file to reflect the changes in dependencies. Running resolver because the following dependencies were added: 'lottie-ios' (https://github.com/airbnb/lottie-ios.git)
Resolved source packages:
  PocketCastsUtils: /opt/ci/builds/builder/automattic/pocket-casts-ios/Modules/Utils
  PocketCastsServer: /opt/ci/builds/builder/automattic/pocket-casts-ios/Modules/Server
  FMDB: https://github.com/ccgus/fmdb.git @ master (dcd5bb6)
  PocketCastsDataModel: /opt/ci/builds/builder/automattic/pocket-casts-ios/Modules/DataModel
  SwiftProtobuf: https://github.com/apple/swift-protobuf.git @ 1.8.0
2024-06-04 21:31:09.548 xcodebuild[1285:8256] Writing error result bundle to /var/folders/gt/3grbfxs13w53jdjmg4f_c6_r0000gn/T/ResultBundle_2024-04-06_21-31-0009.xcresult
xcodebuild: error: Could not resolve package dependencies:
  an out-of-date resolved file was detected at /opt/ci/builds/builder/automattic/pocket-casts-ios/podcasts.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved, which is not allowed when automatic dependency resolution is disabled; please make sure to update the file to reflect the changes in dependencies. Running resolver because the following dependencies were added: 'lottie-ios' (https://github.com/airbnb/lottie-ios.git)
```

See
https://buildkite.com/automattic/pocket-casts-ios/builds/7089#018fe6a8-c33a-40bd-b933-8d35f844205a

**Comment by leandroalonso** [comment]
@mokagio a question out of curiosity: is the prototype build generated for each PR?

**Comment by SergioEstevao** [comment]
@mokagio I'm having issues using the self-serve tool, it looks I wasn't able to join the AppCenter Automattic organization... Can you give it a look?

**Comment by mokagio** [comment]
For reference, the answer to 

> @mokagio a question out of curiosity: is the prototype build generated for each PR?

is in this other PR: https://github.com/Automattic/pocket-casts-ios/pull/1804#issuecomment-2151213221

TL;DR. At the moment, yes. But happy to change it.

**Comment by SergioEstevao** [comment]
Finally I was able to test it, and the build is working great. 
One nice thing to have is to have a different app name or icon to make it easy to distinguish from the production app.
But for now is all working great but as @leandroalonso  said maybe it's better to have a button in CI system to trigger a new test build?

**Comment by leandroalonso** [comment]
> But for now is all working great but as @leandroalonso said maybe it's better to have a button in CI system to trigger a new test build?

We have this now, it was merged as part of #1808

> One nice thing to have is to have a different app name or icon to make it easy to distinguish from the production app.

We can do this ourselves. :D

**Review by leandroalonso — APPROVED** [review]
Thanks for working on that @mokagio!
